src/main/resources/static/data-mapping.js

```diff
@@ -16,122 +23,143 @@ $( document ).on( "mousemove", function( event ) {
-            if(!is_tree_view) {
-                d3.selectAll('.mitem').each(function(d,i){
-                    if(d.connected) {
-                        var m = 5;
-                        d3.selectAll('.griditem').each(function(gridd, gridi) {
-                            if(gridd.name == d.usrname) {
-                                gridd.x = +d.x - 2*m - d.w;
-                                gridd.y = +d.y;
-                                gridd.mapped = true;
-                                gridd.matchingItem = d;
+        clearMapping();
+        for(var i=0; i<map_data.length; i++) {
+            var pair = map_data[i];
+            var ocd = findOCitem(pair['Study_Site'], pair['EventName'], pair['CRFname'], pair['CRFversion'], pair['ocItemName']);
+            var usritem_obj = findUsrItem(pair['usrItemName']);
+            // console.log(ocd);
+
+            if(ocd !== null && usritem_obj !== null) {
+                var ocname = ocd.name;
+                var ocCRFv = ocd.parent.name;
+                var ocCRF = ocd.parent.parent.name;
+                var ocEventName = ocd.parent.parent.parent.name;
+                var ocStudy = ocd.parent.parent.parent.parent.name;
+                var path = ocStudy+"\t"+ocEventName+"\t"+ocCRF+"\t"+ocCRFv+"\t"+ocname;
+                var d = usritem_obj.datum();
+                if(mapped_ocitems.indexOf(path) == -1) {
+                    mapped_ocitems.push(path);
+                    d.mapped = true;
+                    d.ocItemName = ocname;
+                    d.ocCRFv = ocCRFv;
+                    d.ocCRF = ocCRF;
+                    d.ocEventName = ocEventName;
+                    d.ocStudy = ocStudy;
+                    d.ocItemData = ocd;
+                    d.ocPath = path;
+                }
+                positionUsrList(usr_item_data, 800);
+            }
+        }
+
+        function findOCitem(study, event, crf, crfv, itemname) {
+            for(var _study in oc_data.children) {
+                _study = oc_data.children[_study];
+                if(_study.name == study) {
+                    for(var _event in _study.children) {
+                        _event = _study.children[_event];
+                        if(_event.name == event) {
+                            for(var _crf in _event.children) {
+                                _crf = _event.children[_crf];
+                                if(_crf.name == crf) {
+                                    for(var _crfv in _crf.children) {
+                                        _crfv = _crf.children[_crfv];
+                                        if(_crfv.name == crfv) {
+                                            for(var _item in _crfv.children) {
+                                                _item = _crfv.children[_item];
+                                                if(_item.name == itemname) {
+                                                    return _item;
```

Comment:
I think because it is required that an item must have a path consisting of study, event-name, CRF name and CRF version, this nesting code is actually easier to maintain than, say we hide the specific path information by introducing some generic array indexing and tree tracing.